src/node_crypto.cc

```diff
@@ -1025,6 +1066,46 @@ void SecureContext::LoadPKCS12(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+#ifndef OPENSSL_NO_ENGINE
+void SecureContext::SetClientCertEngine(
+  const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  CHECK(args.Length() == 1 && args[0]->IsString());
+
+  SecureContext* sc = Unwrap<SecureContext>(args.This());
+
+  ClearErrorOnReturn clear_error_on_return;
+
+  // If an engine was previously set, free it.
+  // This is because SSL_CTX_set_client_cert_engine does not itself
+  // 'support' multiple calls by cleaning up before overwriting the
+  // client_cert_engine internal context variable
+  if (sc->ctx_->client_cert_engine != NULL) {
+    ENGINE* e = sc->ctx_->client_cert_engine;
+    sc->ctx_->client_cert_engine = nullptr;
+    ENGINE_finish(e);
+  }
+
+  // Load engine
+  const node::Utf8Value engine_id(env->isolate(), args[0]);
+  ENGINE* engine = LoadEngineById(env, *engine_id);
```

Comment:
It would be infinitesimally nicer if `LoadEngineById()` returned the error message instead of throwing an exception.  You could return the error code in an in/out parameter or pass a string buffer like this:
```c++
ENGINE* LoadEngineById(const char* engine_id, char (*errbuf)[1024]) { /* ... */ }
// ...
char errbuf[1024];
ENGINE* engine = LoadEngineById(*engine_id, &errbuf);
if (engine == nullptr)
  return env->ThrowError(errbuf);
```